bubble_sort.rb

```diff
@@ -0,0 +1,34 @@
+def bubble_sort(array)
+  x = 0
+  length = array.length
+  while x < length - 1
+    if !array[x + 1].nil? && array[x] > array[x + 1]
+      array[x], array[x + 1] = array[x + 1], array[x]
+      x = 0
+    end
+    x += 1
+  end
+  p array
+end
```

Comment:
- [ ] If you test this array `p bubble_sort([10,9,8,7,6,5,4,3,2,1])` it gives [9, 1, 2, 3, 4, 5, 6, 7, 8, 10], please check your logic

  **Hint:** About your logic, please notice that the order of the comparison in bubble sort is like this:
![SWYiwVcMCbdsToyWuAZscP5obiz1-3762u3wq4](https://user-images.githubusercontent.com/53256139/94982405-c00a3c80-04ff-11eb-858d-852b57f6507f.gif)
Both your methods process the array like this with each change:
![image](https://user-images.githubusercontent.com/53256139/94985455-93185280-051c-11eb-94c1-0792eb5124f2.png)

👉 And in bubble sort, the bigger number should be accommodated in the last position first